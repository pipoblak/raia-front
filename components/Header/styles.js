import styled from 'styled-components';

export const Wrapper = styled.header`
  width: 100%;
  height: 95px;
  display: flex;
  border-bottom: solid 2px #e2e2e2;
`;

export const Logo = styled.img`
  margin: auto 0;
`;

export const Links = styled.div`
  margin: auto;
  margin-right: 0;
  margin-top: 26px;
`;

export const Link = styled.a`
  color: ${(p) => p.theme.primary};
  text-decoration: none;
  font-size: 20px;
  font-weight: 400;
  margin: 0.5rem;
`;

export default Wrapper;
