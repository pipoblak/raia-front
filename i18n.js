import NextI18Next from 'next-i18next'
const path = require('path')

module.exports = new NextI18Next({
  defaultLanguage: 'ptBR',
  otherLanguages: ['en'],
  localeSubpaths: {},
  localePath: path.resolve('./public/static/locales')
})