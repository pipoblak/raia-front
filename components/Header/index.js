import React from 'react';
import { Wrapper, Logo, Links, Link } from './styles';
import { withTheme } from 'styled-components';
import PropTypes from 'prop-types';
import themes from '../../styles/themes';

const Header = () => {
  return (
    <Wrapper data-testid="header">
      <Logo src={require('../../public/images/rd.png')} />
      <Links>
        <Link href="#">HTML5</Link>
        <Link href="#">CSS3</Link>
        <Link href="#">JAVASCRIPT</Link>
        <Link href="#">REACT</Link>
        <Link href="#">REDUX</Link>
      </Links>
    </Wrapper>
  );
};

Header.propTypes = {
  theme: PropTypes.object.isRequired
};

Header.defaultProps = {
  theme: themes.default
};

export default withTheme(Header);
