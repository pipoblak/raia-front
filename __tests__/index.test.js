import { render, screen } from '@testing-library/react';
import Home from '../pages/index';
import Header from '../components/Header';
import { appWithTranslation } from '../i18n';

describe('Home', () => {
  var I18nApp;
  beforeEach(async () => {
    const props = await Home.getInitialProps();
    I18nApp = appWithTranslation(Home);
  });

  it('renders without crashing', () => {
    const { container } = render(<I18nApp />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  describe('Header', () => {
    it('renders', () => {
      const { container, getByTestId } = render(<I18nApp />);
      expect(getByTestId('header').innerHTML).toMatchSnapshot();
    });
  });
});
