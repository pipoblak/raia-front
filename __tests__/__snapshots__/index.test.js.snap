// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Header renders 1`] = `"<img src=\\"/images/rd.png\\" class=\\"styles__Logo-sc-198cq94-1 dzztZJ\\"><div class=\\"styles__Links-sc-198cq94-2 iEyVpx\\"><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">HTML5</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">CSS3</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">JAVASCRIPT</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">REACT</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">REDUX</a></div>"`;

exports[`Home renders without crashing 1`] = `"<div class=\\"styles__Wrapper-sc-96wnlc-0 dcsTEj\\"><div class=\\"styles__Container-sc-96wnlc-1 fPSALL\\"><header data-testid=\\"header\\" class=\\"styles__Wrapper-sc-198cq94-0 jyYpTB\\"><img src=\\"/images/rd.png\\" class=\\"styles__Logo-sc-198cq94-1 dzztZJ\\"><div class=\\"styles__Links-sc-198cq94-2 iEyVpx\\"><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">HTML5</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">CSS3</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">JAVASCRIPT</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">REACT</a><a href=\\"#\\" class=\\"styles__Link-sc-198cq94-3 hUdcYn\\">REDUX</a></div></header></div></div>"`;
